### ISSUE TITLE
Rule S2437: FP with local fields

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4774]
### Description

Rule S2437 is raising false positives when one of the operators is a local field initialized with an expected value (0 or -1).

### Repro steps

```
class LocalFields
{
    private int start = 0;

    public void UpdateStart(int val) => start = val;

    public override int GetHashCode()
    {
        return Method() ^ start; // Noncompliant - FP: start is a local field which can be updated to have a value != 0

        int Method() => 42;
    }
}
```

### Expected behavior

The issue should not be raised.

### Actual behavior

Issue is raised on the line with `Noncompliant` comment.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins version 8.26
